In [19]:
import pandas as pd

In [20]:
train_slog = pd.read_csv("train-SLOG.tsv", names=["sent", "rep", "type"], sep="\t")
train_cogs = pd.read_csv("train-COGS.tsv", names=["sent", "rep", "type"], sep="\t")
gen_slog = pd.read_csv("gen-SLOG.tsv", names=["sent", "rep", "type"], sep="\t")
gen_cogs = pd.read_csv("gen-COGS.tsv", names=["sent", "rep", "type"], sep="\t")

train_slog

sent  \
0                   A cake was cooked by the scientist .   
1                            A cake was eaten by a boy .   
2                                            A fly ran .   
3         A hen gave the flower beside a chair to Liam .   
4        A mouse was given a cake on the table by Liam .   
...                                                  ...   
32750  Victoria ate the donut on the crack in the hou...   
32751         A cat that admired that Emily ate the cake   
32752                                           tolerate   
32753                 A queen that the sailor worshipped   
32754                            Who did a sailor like ?   

                                                     rep               type  
0      * scientist ( x _ 6 ) ; cake ( x _ 1 ) AND coo...    in_distribution  
1      cake ( x _ 1 ) AND eat . theme ( x _ 3 , x _ 1...    in_distribution  
2        fly ( x _ 1 ) AND run . agent ( x _ 2 , x _ 1 )    in_distribution  
3      * flower ( x _ 4 ) ; hen ( x _ 1 ) AND give . ...    in_distribution  
4      * table ( x _ 8 ) ; mouse ( x _ 1 ) AND give ....    in_distribution  
...                                                  ...                ...  
32750  * donut ( x _ 3 ) ; * crack ( x _ 6 ) ; * hous...               pp_4  
32751  * cake ( x _ 8 ) ; cat ( x _ 1 ) AND cat . nmo...   standalone_NP_RC  
32752  LAMBDA a . LAMBDA b . LAMBDA e . tolerate . ag...          primitive  
32753  * sailor ( x _ 4 ) ; queen ( x _ 1 ) AND queen...   standalone_NP_RC  
32754  sailor ( x _ 3 ) AND like . agent ( x _ 4 , x ...  wh_Q_simple_trans  

[32755 rows x 3 columns]

In [21]:
train_cogs

sent  \
0                       A rose was helped by a dog .   
1                          The sailor dusted a boy .   
2                            Emma rolled a teacher .   
3                           Evelyn rolled the girl .   
4        A cake was forwarded to Levi by Charlotte .   
...                                              ...   
24150   Emma gave a cookie beside the bed to a boy .   
24151                           The captain burned .   
24152         The donut was appreciated by a guard .   
24153      The brush was given to Emma by the girl .   
24154  A drink was lended to a landlord by the pig .   

                                                     rep             type  
0      rose ( x _ 1 ) AND help . theme ( x _ 3 , x _ ...  in_distribution  
1      * sailor ( x _ 1 ) ; dust . agent ( x _ 2 , x ...  in_distribution  
2      roll . agent ( x _ 1 , Emma ) AND roll . theme...  in_distribution  
3      * girl ( x _ 3 ) ; roll . agent ( x _ 1 , Evel...  in_distribution  
4      cake ( x _ 1 ) AND forward . theme ( x _ 3 , x...  in_distribution  
...                                                  ...              ...  
24150  * bed ( x _ 6 ) ; give . agent ( x _ 1 , Emma ...  in_distribution  
24151  * captain ( x _ 1 ) ; burn . theme ( x _ 2 , x...  in_distribution  
24152  * donut ( x _ 1 ) ; appreciate . theme ( x _ 3...  in_distribution  
24153  * brush ( x _ 1 ) ; * girl ( x _ 8 ) ; give . ...  in_distribution  
24154  * pig ( x _ 9 ) ; drink ( x _ 1 ) AND lend . t...  in_distribution  

[24155 rows x 3 columns]

The SLOG dataset is built on top of the COGS training set, which contains $24,155$ unique sentences. In theory, this should mean that all sentences from the original COGS should be present within SLOG (?); however, this does not seem to be the case when we take a look at the set difference:

In [40]:
len(set(train_cogs["sent"]) - set(train_slog["sent"]))

1155

We can take a look at ten of the sentences present in COGS that seem to be "missing" from SLOG:

In [41]:
list(set(train_cogs["sent"]) - set(train_slog["sent"]))[:10]

['The cat was lended a towel beside a table by Luke .',
 'Noah expected that William lended a child a pillow .',
 'A mouse lended a rose on the stage to Liam .',
 'The bunny lended the cake in a puddle to the girl .',
 'A girl lended Olivia the donut .',
 'The banana was lended to Chloe by Carter .',
 'A pizza was lended to a guest .',
 'A scientist lended a cat a donut beside a computer .',
 'A monkey was lended the cake beside a stage .',
 'Emma lended the chicken the cake on a road .']

Furthermore, the "missing" entries seem to affect only `in_distribution` examples:

In [43]:
train_slog[train_slog["type"] == "in_distribution"]

sent  \
0                 A cake was cooked by the scientist .   
1                          A cake was eaten by a boy .   
2                                          A fly ran .   
3       A hen gave the flower beside a chair to Liam .   
4      A mouse was given a cake on the table by Liam .   
...                                                ...   
32741                                    Emma packed .   
32745                  The cake was found by the boy .   
32746          The fish cleaned the cake in a drawer .   
32748                       Benjamin longed to paint .   
32749      A cat was given the cookie beside a sword .   

                                                     rep             type  
0      * scientist ( x _ 6 ) ; cake ( x _ 1 ) AND coo...  in_distribution  
1      cake ( x _ 1 ) AND eat . theme ( x _ 3 , x _ 1...  in_distribution  
2        fly ( x _ 1 ) AND run . agent ( x _ 2 , x _ 1 )  in_distribution  
3      * flower ( x _ 4 ) ; hen ( x _ 1 ) AND give . ...  in_distribution  
4      * table ( x _ 8 ) ; mouse ( x _ 1 ) AND give ....  in_distribution  
...                                                  ...              ...  
32741                      pack . agent ( x _ 1 , Emma )  in_distribution  
32745  * cake ( x _ 1 ) ; * boy ( x _ 6 ) ; find . th...  in_distribution  
32746  * fish ( x _ 1 ) ; * cake ( x _ 4 ) ; clean . ...  in_distribution  
32748  long . agent ( x _ 1 , Benjamin ) AND long . x...  in_distribution  
32749  * cookie ( x _ 5 ) ; cat ( x _ 1 ) AND give . ...  in_distribution  

[24000 rows x 3 columns]

In [44]:
train_cogs[train_cogs["type"] == "in_distribution"]

sent  \
0                       A rose was helped by a dog .   
1                          The sailor dusted a boy .   
2                            Emma rolled a teacher .   
3                           Evelyn rolled the girl .   
4        A cake was forwarded to Levi by Charlotte .   
...                                              ...   
24150   Emma gave a cookie beside the bed to a boy .   
24151                           The captain burned .   
24152         The donut was appreciated by a guard .   
24153      The brush was given to Emma by the girl .   
24154  A drink was lended to a landlord by the pig .   

                                                     rep             type  
0      rose ( x _ 1 ) AND help . theme ( x _ 3 , x _ ...  in_distribution  
1      * sailor ( x _ 1 ) ; dust . agent ( x _ 2 , x ...  in_distribution  
2      roll . agent ( x _ 1 , Emma ) AND roll . theme...  in_distribution  
3      * girl ( x _ 3 ) ; roll . agent ( x _ 1 , Evel...  in_distribution  
4      cake ( x _ 1 ) AND forward . theme ( x _ 3 , x...  in_distribution  
...                                                  ...              ...  
24150  * bed ( x _ 6 ) ; give . agent ( x _ 1 , Emma ...  in_distribution  
24151  * captain ( x _ 1 ) ; burn . theme ( x _ 2 , x...  in_distribution  
24152  * donut ( x _ 1 ) ; appreciate . theme ( x _ 3...  in_distribution  
24153  * brush ( x _ 1 ) ; * girl ( x _ 8 ) ; give . ...  in_distribution  
24154  * pig ( x _ 9 ) ; drink ( x _ 1 ) AND lend . t...  in_distribution  

[24000 rows x 3 columns]

In [42]:
set1 = set(train_cogs[train_cogs["type"] == "in_distribution"]["sent"])
set2 = set(train_slog[train_slog["type"] == "in_distribution"]["sent"])

only_in_df1 = set1 - set2
only_in_df2 = set2 - set1

print("in_dist sents only in COGS:", len(only_in_df1))
print("in_dist sents only in SLOG:", len(only_in_df2))

in_dist sents only in COGS: 1155
in_dist sents only in SLOG: 1155


So it seems that $1,155$ in-distribution sentences have been modified/removed between COGS and SLOG?